apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/R4ExplanationOfBenefitResourceProviderTest.java

```diff
@@ -219,21 +239,86 @@ public void setup() {
     when(metricRegistry.timer(any())).thenReturn(mockTimer);
     when(mockTimer.time()).thenReturn(mockTimerContext);
 
-    // transformer mocking
-    when(carrierClaimTransformer.transform(any(), anyBoolean())).thenReturn(testEob);
-    when(inpatientClaimTransformer.transform(any())).thenReturn(testEob);
-    when(outpatientClaimTransformer.transform(any())).thenReturn(testEob);
-    when(partDEventTransformer.transform(any())).thenReturn(testEob);
-    when(hhaClaimTransformer.transform(any())).thenReturn(testEob);
-    when(snfClaimTransformer.transform(any())).thenReturn(testEob);
-    when(hospiceClaimTransformer.transform(any())).thenReturn(testEob);
-    when(dmeClaimTransformer.transform(any(), anyBoolean())).thenReturn(testEob);
+    // NPI and FDA drug mocking
+    when(npiOrgLookup.retrieveNPIOrgDisplay(Optional.empty())).thenReturn(Optional.empty());
 
     setupLastUpdatedMocks();
 
     mockHeaders();
   }
 
+  /** Sets up claim transformers. */
+  private void setupTransformers() {
+    carrierClaimTransformer =
+        new CarrierClaimTransformerV2(metricRegistry, drugDisplayLookup, npiOrgLookup);
+    partDEventTransformer = new PartDEventTransformerV2(metricRegistry, drugDisplayLookup);
+    dmeClaimTransformer = new DMEClaimTransformerV2(metricRegistry, drugDisplayLookup);
+    hhaClaimTransformer = new HHAClaimTransformerV2(metricRegistry, npiOrgLookup);
+    hospiceClaimTransformer = new HospiceClaimTransformerV2(metricRegistry, npiOrgLookup);
+    inpatientClaimTransformer = new InpatientClaimTransformerV2(metricRegistry, npiOrgLookup);
+    outpatientClaimTransformer =
+        new OutpatientClaimTransformerV2(metricRegistry, drugDisplayLookup, npiOrgLookup);
+    snfClaimTransformer = new SNFClaimTransformerV2(metricRegistry, npiOrgLookup);
```

Comment:
So now we need to delve into the realm of mocking a transformer constructor; to do that we may need extensions like PowerMock or some other mechanism that allow us to invoke some method that handles the instantiation of the transformer. Open to suggestions here, but to me that LOE is beyond the AC for this ticket which is  enale parallel processing of transformers.

Reply:
I'll pull down this branch and see if what I'm thinking is more complicated than I think, but mocking the transformers _shouldnt_ be too onerous.

Reply:
Per discussion you shouldnt really even need the transformers at all in this test, since the transformers are logic thats within the new Future task to read/transform the claims data. Ideally you should only need to mock those task classes, and what comes back from them, and we dont need the mock transformers here at all.